File: Frontend/src/lib/services/strategies-overview.ts
import { invoke } from "@tauri-apps/api/core";
import { Result } from "@/types/result";
import { StrategyOverview } from "@/interfaces/strategy-overview";
import { Strategy } from "@/interfaces/entities/strategy";

async function select_strategies_overview(
    env: string,
    query: Strategy
): Promise<Result<StrategyOverview[]>> {
    try {
        const data = await invoke<StrategyOverview[]>(
            "select_strategies_overview",
            {
                env,
                query,
            }
        );

        return { ok: true, data };
    } catch (error) {
        return { ok: false, error };
    }
}

export const strategyOverviewService = {
    select: (
        env: string,
        query: Strategy
    ): Promise<Result<StrategyOverview[]>> =>
        select_strategies_overview(env, query),
};

File: Frontend/src-tauri/src/command/strategies_overview.rs
use crate::core::select_strategies_overview_core;
use crate::models::structs::StrategyRequest;

#[tauri::command]
pub async fn select_strategies_overview(
    env: String,
    query: StrategyRequest,
) -> Result<Vec<crate::models::structs::StrategyOverview>, String> {
    select_strategies_overview_core(env, query).await
}
